File: examples/direct/TextProducer.cpp
#include "JobPool.h"

#include <protocol/TBinaryProtocol.h>
#include <transport/TSocket.h>
#include <transport/TTransportUtils.h>

#include <iostream>

using namespace jp;

using namespace apache::thrift;
using namespace apache::thrift::protocol;
using namespace apache::thrift::transport;

using boost::shared_ptr;

using namespace std;

int main(int argc, char** argv)
{
	shared_ptr<TTransport> socket(new TSocket("localhost", 9090));
	shared_ptr<TTransport> transport(new TBufferedTransport(socket));
	shared_ptr<TProtocol> protocol(new TBinaryProtocol(transport));
	JobPoolClient client(protocol);
	transport->open();

	cout << "Adding a chicken..." << endl;
	client.add("text", "chicken");
	cout << "Added a chicken." << endl;
	return 0;
}

File: examples/direct/TextProducer.java
import org.apache.thrift.TException;
import org.apache.thrift.transport.TTransport;
import org.apache.thrift.transport.TSocket;
import org.apache.thrift.transport.TTransportException;
import org.apache.thrift.protocol.TBinaryProtocol;
import org.apache.thrift.protocol.TProtocol;
import uk.co.fredemmott.jp.*;

public class TextProducer
{
	public static void main(String[] args)
	{
		try
		{
			TTransport transport = new TSocket("localhost", 9090);
			TProtocol protocol = new TBinaryProtocol(transport);
			JobPool.Client client = new JobPool.Client(protocol);

			transport.open();

			System.out.println("Adding a bean...");
			client.add("text", "bean".getBytes("UTF-8"));
			System.out.println("Added a bean.");
		}
		catch (java.io.UnsupportedEncodingException e)
		{
			System.out.println("UTF-8 encoding not supported, couldn't add item.");
		}
		catch (NoSuchPool e)
		{
			System.out.println("Pool 'text' does not exist, couldn't add item.");
		}
		catch (TException e)
		{
			e.printStackTrace();
		}
	}
}
